source/reference/minio-server/minio-server.rst

```diff
@@ -203,39 +239,72 @@ Root Credentials
 
 .. envvar:: MINIO_ACCESS_KEY_OLD
 
-   Used for rotating the :ref:`root <minio-users-root>` user access
-   key.
+   .. deprecated:: RELEASE.2021-04-22T15-44-28Z
 
-   Restart the :mc:`minio server` process with *all* of the following
-   environment variables to rotate the root credentials:
+   To perform root credential rotation, modify the
+   :envvar:`MINIO_ROOT_USER_FILE` and `MINIO_ROOT_PASSWORD_FILE` environment
+   variables.
 
-   - :envvar:`MINIO_ACCESS_KEY_OLD` set to the old access key.
-   - :envvar:`MINIO_ACCESS_KEY` set to the new access key.
-   - :envvar:`MINIO_SECRET_KEY_OLD` set to the old secret key.
-   - :envvar:`MINIO_SECRET_KEY` set to the new secret key.
+.. envvar:: MINIO_SECRET_KEY_OLD
 
-   The :mc:`minio server` process automatically detects and re-encrypts 
-   the server configuration with the new credentials. After the process
-   restarts successfully, you can restart it without 
-   :envvar:`MINIO_ACCESS_KEY_OLD`.
+   .. deprecated:: RELEASE.2021-04-22T15-44-28Z
 
-.. envvar:: MINIO_SECRET_KEY_OLD
+   To perform root credential rotation, modify the
+   :envvar:`MINIO_ROOT_USER_FILE` and `MINIO_ROOT_PASSWORD_FILE` environment
+   variables.
+
+Key Management Service and Encryption
```

Comment:
@aead this is somewhat based on a very old KES PR I need to review, so let me know how far off base I am here.